data_filter_azure/README.containerregistry.md

```diff
@@ -0,0 +1,108 @@
+# Data Filter Example with Data from Azure table storage
+
+Usually, OPA keeps policies and data in-memory. But in cases where the context data cannot be loaded, there are sevaral options to support. This directory contains a sample where the data and policies are downloaded as bundles using the [OPA's bundle](https://www.openpolicyagent.org/docs/latest/bundles/) feature. The polcies and data are loaded on the fly and enforced by OPA. Currently OPA is configured to download one bundle from a single source. These bundles are stored in Azure container registry which is OCI-compatible registry. You can read more about this in  [the post](https://stevelasker.blog/2019/01/25/cloud-native-artifact-stores-evolve-from-container-registries/). This sample contains a sample server that uses OPA's Get Document API on the bundle that is configured with it. The server also has an API that downloads the bundle from container registry. OPA is configured to download bundle from this HTTP server and enforce policy using it.
```

Comment:
Reading this I struggled to understand what the code illustrates.  Is it (i) a bundle server or (ii) bundles served by Azure container registry.  And what does "OPA's GET Document API" have to do with bundles?  OPA's GET API is called by clients of OPA; OPA does not call GET Document to download bundles.

Please clarify the intent of this PR.  What part is a code contribution and what part is a placeholder for what the user must provide?  It looks like the bundle name/path is configured via environmental variables (instead of as part of OPA's bundle API configuration, which is where you specify the bundle name).

Reply:
@timothyhinrichs  Sorry for the late reply. I was on vacation for the last few weeks.  

OPA can be configured to download the bundles from a web server. Here the bundle server is not a web server, it is a standard OCI compatible registry. There are differences in how bundles are requested from a simple web server versus registry. For example OPA does a GET on the URL to fetch the bundles. On the other handle for OCI compatible registry it uses the API spec of containerd to pull the bundle. As OPA doesn't support downloading bundles directly from registry, it needs some extra steps. 

The PR has two parts 

Initial setup:
1. Setting up azure container registry as the bundle server so that bundles can be pulled from it.
2. Since the OPA needs to have a web server for bundles, a local web server (using flask)  is configured to serve bundles. This web server in turn pulls the bundles from ACR and returns them to OPA

Check access
1. The rest is  similar to other samples listed under the https://github.com/open-policy-agent/contrib/blob/master/data_filter_azure It uses the OPA Get Document API on the bundle to check access for the given request. The OPA related data and policies are in the bundle and the GET document API queries for the specific policy within it. 

Regarding the environment variables for the bundles, these are used by the web server to pull the bundle from the registry.

Reply:
That description is much clearer.  Could you update the paragraph above in the README to say basically that?